report/src/index.ts

```diff
@@ -0,0 +1,357 @@
+            if(cc > colCount){
+                colCount = cc;
+            }
+        });
+        element.classed("report-col-count-"+colCount, true);
+
+    }
+
+    enterDataRow(row, i, ext){
+        const y = i * config.rowHeight;
+
+        let c = 0;
+        this.setContent(y, c++, getAttributeDescWidget(row), undefined, config.rowHeight);
+        this.setContent(y, c++, getStatsWidget(row, ext.statsDataWidth), undefined, config.rowHeight);
+        if(this._showBreakdownColumn)this.setContent(y, c++, getBreakdownWidget(row), undefined, config.rowHeight);
+        if(this._showPopularPatternsColumn)this.setContent(y, c++, getPopularPatternsWidget(row), undefined, config.rowHeight);
+        if(this._showQuartileColumn)this.setContent(y, c++, getQuartileWidget(row), undefined, config.rowHeight);
+
+        return c;
+
+        function getAttributeDescWidget(row) {
+            const p = 8;
+            const b = 1;
+            const w1 = new Html()
+                .html(`<span style="
+                    color:${config.primaryColor};
+                    padding:${p}px;
+                    display:inline-block;
+                    font-size:${config.secondaryFontSize}px;
+                    margin-top:4px;
+                    border:${b}px solid ${config.secondaryColor};
+                    border-radius:4px;
+                    background-color: ${config.offwhiteColor};
+                    width: calc(100% - ${(p * 2) + (b * 2)}px);
+                ">
+                    <i style="
+                        font-size:${config.secondaryFontSize}px;
+                        color:${config.blueColor};
+                    " class="fa ${row.given_attribute_type.slice(0, 6) === "string" ? "fa-font" : "fa-hashtag"}"></i>
+                    <b>${row.attribute}</b>
+                    <span style="float:right;">${row.given_attribute_type}</span>
+                </span>
+                <span style="padding:12px 2px;display:inline-block;font-family: Verdana; color: rgb(51, 51, 51); font-weight: bold; font-size: 14px;">
+                    Optimal:
+                </span>
+                <span style="
+                    color:${config.primaryColor};
+                    padding:4px 8px;
+                    display:inline-block;
+                    font-size:${config.secondaryFontSize}px;
+                    margin-top:4px;
+                    border:1px solid ${config.secondaryColor};
+                    border-radius:4px;
+                    background-color: ${config.offwhiteColor};
+                    float:right;
+                ">
+                    <i style="
+                        font-size:${config.secondaryFontSize}px;
+                        color:${config.blueColor};
+                    " class="fa ${row.best_attribute_type.slice(0, 6) === "string" ? "fa-font" : "fa-hashtag"}"></i>
+                    ${row.best_attribute_type}
+                </span>
+                `)
```

Comment:
What is this displaying?  Should it be a widget?

Reply:
It's the HTMLBadge. There's a PR for it, but we were discussing whether or not it should essentially just be replaced with common_Entity.